docs/apis/core/dml/delegated-transactions.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Delegated database transactions
+tags:
+  - core_dml
+  - DML
+  - core
+  - API
+---
+
```

Comment:
I'd suggest adding an introduction (no title please) before the general principles. Nothing too fancy, just something like:

```suggestion

Moodle allows data manipulation to take place within a database transaction, known as a _Delegated transaction_. This allows you to perform CRUD operations, and roll them back if a failure takes place.

```

Reply:
I've added the text that you have here.